core/common/test/files/SmokeFileTest.kt

```diff
@@ -336,7 +336,30 @@ class SmokeFileTest {
         }
         assertEquals("second third",
             SystemFileSystem.source(path).buffered().use { it.readString() })
+    }
+
+    @Test
+    fun resolve() {
```

Comment:
I think I need to add more tests: to check for an exception if the path does not exist, to check for an absolute path, etc.

Reply:
Added the test for absolute path, all other cases are already there.

It would be nice to test symlink resolution, but we don't have an API for links yet.